addons/l10n_in_withholding/wizard/l10n_in_account_withhold_wizard.py

```diff
@@ -0,0 +1,303 @@
+    def _compute_display_amount_warning(self):
+        for wizard in self:
+            if wizard.related_move_id:
+                wizard.display_amount_warning = wizard.related_move_id.amount_untaxed < sum(line.base for line in wizard.withhold_line_ids)
+            else:
+                wizard.display_amount_warning = wizard.related_payment_id.amount < sum(line.base for line in wizard.withhold_line_ids)
+
+    # ===== MOVE CREATION METHODS =====
+    def action_create_and_post_withhold(self):
+        self.ensure_one()
+        withholding_account_id = self.company_id.l10n_in_withholding_account_id
+        self._validate_withhold_data_on_post(withholding_account_id)
+
+        # Withhold creation and posting
+        vals = self._prepare_withhold_header()
+        move_lines = self._prepare_withhold_move_lines(withholding_account_id)
+        vals['line_ids'] = [Command.create(line) for line in move_lines]
+        withhold = self.env['account.move'].create(vals)
+        withhold.action_post()
+
+        # If the withhold is created from a payment, there is no need to reconcile
+        if not self.related_payment_id:
+            wh_reconc = withhold.line_ids.filtered(
+                lambda l: l.account_id.account_type in ('asset_receivable', 'liability_payable'))
+            inv_reconc = self.related_move_id.line_ids.filtered(
+                lambda l: l.account_id.account_type in ('asset_receivable', 'liability_payable') and not l.reconciled)
+            (inv_reconc + wh_reconc).reconcile()
+        related_record = self.related_move_id or self.related_payment_id
+        withhold.message_post(
+            body=Markup("%s %s: <a href='#' data-oe-model='%s' data-oe-id='%s'>%s</a>") % (
+                _("TDS created from"),
+                self.type_name,
+                related_record._name,
+                related_record.id,
+                related_record.name
+            ))
+        return withhold
+
+    def _prepare_withhold_header(self):
+        """
+        Prepare the header for the withhold entry
+        """
+        vals = {
+            'date': self.date,
+            'journal_id': self.journal_id.id,
+            'partner_id': self.partner_id.id,
+            'move_type': 'entry',
+            'ref': self.reference,
+            'l10n_in_is_withholding': True,
+            'l10n_in_withholding_ref_move_id': self.related_move_id.id or self.related_payment_id.move_id.id,
+        }
+        return vals
+
+    def _prepare_withhold_move_lines(self, withholding_account_id):
+        """
+        Prepare the move lines for the withhold entry
+        """
+        def append_vals(quantity, price_unit, debit, credit, account_id, tax_ids):
+            return {
+                'quantity': quantity,
+                'price_unit': price_unit,
+                'debit': debit,
+                'credit': credit,
+                'account_id': account_id.id,
+                'tax_ids': tax_ids,
```

Comment:
What about amount_currency?  If the withhold is always created in INR then we do not need the complexity in https://github.com/odoo/odoo/pull/168005/files#diff-c65a50add7f512ff449901dec410d569ddaef7486f66a3272baaf2e38ffe9743R15